### PR TITLE
Keep the AsyncMultiMap cache in sync even after split-brains 

### PR DESCRIPTION
**Pull request by Andy2003**
Keep the AsyncMultiMap cache in sync even after split-brains
**does not solve merge conflicts of MultiMap itself**

### CODE REVIEW

**Review by mirkoseifert — APPROVED** [review]
From my point of view, the changes look fine.

File: vertx-hazelcast/src/test/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMapTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.instance.HazelcastTestUtil.closeConnectionBetween;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author <a href="mailto:andreas.berger@kiwigrid.com">Andreas Berger</a>
+ */
+public class HazelcastAsyncMultiMapTest {
```
Comment (jekkel) [inline vertx-hazelcast/src/test/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMapTest.java:32]: It seems you're only testing that hz multimap does not merge "correctly". Shouldn't you test for the clearance of the cache, should you? I thought clearing the cache is what this PR is all about.

File: vertx-hazelcast/src/main/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMap.java [hunk 2]
```diff
@@ -69,14 +77,6 @@ public Void perform() {
             map.remove(entry.getKey(), v);
           }
         }
-        // remove directly from cache as well, b/c if the hazelcast does know the server id , it will never get thrown
```
Comment (jekkel) [inline vertx-hazelcast/src/main/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMap.java:72]: I feel a lot better with this going out again. This is what we added a version before, right?

**Review by jekkel — CHANGES_REQUESTED** [review]
Looks good overall - only a test for the changed behaviour should be added.

File: vertx-hazelcast/src/main/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMap.java (hunk 2 above)
Comment (Andy2003) [reply vertx-hazelcast/src/main/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMap.java:72]: yes

File: vertx-hazelcast/src/test/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMapTest.java (hunk 1 above)
Comment (Andy2003) [reply vertx-hazelcast/src/test/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMapTest.java:32]: I tested the changed behaviour: if you remove the LifeCycleListener the test will fail. I just test that we get the same values which are stored in the cluster.

**Pull request merged**